python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -271,20 +344,38 @@ def runQuantum(self, task, quantum, taskDef, butler):
         # Get the input and output references for the task
         inputRefs, outputRefs = taskDef.connections.buildDatasetRefs(quantum)
 
-        # Call task runQuantum() method. Any exception thrown by the task
-        # propagates to caller.
-        task.runQuantum(butlerQC, inputRefs, outputRefs)
+        # Call task runQuantum() method.  Catch a few known failure modes and
+        # translate them into specific
+        try:
+            task.runQuantum(butlerQC, inputRefs, outputRefs)
+        except NoWorkFound as err:
+            # Not an error, just an early exit.
+            _LOG.info("Task '%s' on quantum %s exited early: %s",
+                      taskDef.label, quantum.dataId, str(err))
+            pass
+        except RepeatableQuantumError as err:
+            if self.exitOnKnownError:
+                _LOG.warning("Caught repeatable quantum error for %s (%s):", taskDef, quantum.dataId)
+                _LOG.warning(err, exc_info=True)
+                sys.exit(err.EXIT_CODE)
+            else:
+                raise
+        except InvalidQuantumError as err:
+            _LOG.fatal("Invalid quantum error for %s (%s): %s", taskDef, quantum.dataId)
+            _LOG.fatal(err, exc_info=True)
+            sys.exit(err.EXIT_CODE)
```

Comment:
Should this be the same exit code here?

Reply:
I don't understand.  Same exit code as what?

Reply:
Never mind, I clarified my question on the pipe_base review